src/web/src/components/Posts/Post.tsx

```diff
@@ -286,6 +286,39 @@ const extractBlogClassName = (url: string) => {
 // a 'guid' from a YouTube video is usually written as 'yt:video:id'
 const extractVideoId = (post: Post): string => post.guid.split(':')[2];
 
+function handleZoom(e: MouseEvent) {
+  let canZoom = 1;
+  // Zoom out of all the currently zoomed images on click
+  document.querySelectorAll<HTMLElement>('.zoomed-image').forEach((zoomedImg) => {
+    zoomedImg.classList.remove('zoomed-image');
+    // if the clicked element is a zoomed image, do not allow further zooming
+    if (zoomedImg === e.target) canZoom = 0;
+  });
+  // Check if the clicked element is an image or an anchor that contains an image
+  let image = null;
+  if (e.target instanceof HTMLImageElement) {
+    image = e.target;
+  } else if (
+    e.target instanceof HTMLAnchorElement &&
+    e.target.firstChild instanceof HTMLImageElement
+  ) {
+    image = e.target.firstChild;
+  }
+  // Zoom in on the image if it is not already zoomed
+  if (image) {
+    e.preventDefault();
+    // and it can be zoomed
+    if (canZoom) {
+      // calculate the scale factor so that the image fits the telescope-post-content section's width
+      const contentContainer = image.closest('.telescope-post-content');
```

Comment:
Wow.

I'm just thinking about how it would look if you fit this image to the view height,  instead of width of the post content.

Reply:
it uses view height and width now instead of the post content.

Reply:
Love the new approach